src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.SdkArchiveDiff/GetClosestArchive.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+
+public abstract class GetClosestArchive : Microsoft.Build.Utilities.Task, ICancelableTask
+{
+    [Required]
+    public required string BuiltArchivePath { get; init; }
+
+    [Output]
+    public string ClosestOfficialArchivePath { get; set; } = "";
+
+    private string? _builtVersion;
+    protected string BuiltVersion
+    {
+        get => _builtVersion ?? throw new InvalidOperationException();
+        private set => _builtVersion = value;
+    }
+
+    private string? _builtRid;
+    protected string BuiltRid
+    {
+        get => _builtRid ?? throw new InvalidOperationException();
+        private set => _builtRid = value;
+    }
+
+    private string? _archiveExtension;
+    protected string ArchiveExtension
+    {
+        get => _archiveExtension ?? throw new InvalidOperationException();
+        private set => _archiveExtension = value;
+    }
+
+    /// <summary>
+    /// The name of the package to find the closest official archive for. For example, "dotnet-sdk" or "aspnetcore-runtime".
+    /// </summary>
+    protected abstract string ArchiveName { get; }
+
+    private CancellationTokenSource _cancellationTokenSource = new();
+    protected CancellationToken CancellationToken => _cancellationTokenSource.Token;
+    public void Cancel()
+    {
+        _cancellationTokenSource.Cancel();
+    }
+
+    /// <summary>
+    /// Get the URL of the latest official archive for the given version string and RID.
+    /// </summary>
+    public abstract Task<string?> GetLatestOfficialArchiveUrl();
```

Comment:
Yeah, that's true, but since ideally this would eventually get the installer commit or parse the build version and find the best build that's not necessarily the latest I'm leaning towards keeping "Closest".